com.unity.shadergraph/Editor/Generation/Processors/PropertyCollector.cs

```diff
@@ -14,26 +16,186 @@ public struct TextureInfo
             public bool modifiable;
         }
 
-        public readonly List<AbstractShaderProperty> properties = new List<AbstractShaderProperty>();
+        private Target m_Target = null;
+        private UInt64 m_DeclarerFlag = (UInt64)DeclarerFlags._NodeOrGraph;
 
-        public void AddShaderProperty(AbstractShaderProperty chunk)
+        private bool m_ReadOnly;
+        private Target m_HLSLPropertyTarget = null;
+        private List<HLSLProperty> m_HLSLProperties = null;
+
+        enum DeclarerFlags
+        {
+            _NodeOrGraph = (1 << 0),    // property declared by a graph (graph property) or a node (or slot on the node)
+            _Target = (1 << 1),         // all bits starting with this one represent properties added by targets in the Target list
+        }
+
+        private struct Property
+        {
+            public AbstractShaderProperty property;
+            public UInt64 declarerFlags;
+        }
+
+        private List<Target> m_Targets = new List<Target>();
+
+        // reference name ==> property index in list
+        Dictionary<string, int> m_ReferenceNames = new Dictionary<string, int>();
+
+        // list of properties (kept in a list to maintain deterministic declaration order)
+        private List<Property> m_Properties = new List<Property>();
+
+        public int propertyCount => m_Properties.Count;
+        public IEnumerable<AbstractShaderProperty> properties => m_Properties.Select(p => p.property);
+        public AbstractShaderProperty GetProperty(int index) { return m_Properties[index].property; }
+
+        public void Sort()
+        {
+            m_Properties.Sort((a, b) => String.CompareOrdinal(a.property.referenceName, b.property.referenceName));
+        }
+
+        public void SetReadOnly()
+        {
+            m_ReadOnly = true;
+        }
+
+        public void BeginTargetCollection(Target target)
+        {
+            // Debug.Log("Begin Collection for Target " + target.displayName);
```

Comment:
Did you mean to leave this in here (will it come up often enough that a developer can't add just add the log call during debugging)?

Reply:
I will remove it.